apps/rush-lib/src/cli/actions/PublishAction.ts

```diff
@@ -390,12 +385,16 @@ export class PublishAction extends BaseRushAction {
 
   private _packageExists(packageConfig: RushConfigurationProject): boolean {
     const env: { [key: string]: string | undefined } = PublishUtilities.getEnvArgs();
-    if (this._registryUrl.value) {
-      env['npm_config_registry'] = this._registryUrl.value; // tslint:disable-line:no-string-literal
+    const registry: string = this._getRegistryAndUpdateEnv(env);
+    let args: string = '';
+    if (this._npmAuthToken.value) {
+      args = this._getAuthTokenArg(registry);
     }
```

Comment:
It'd be useful to extract this into a single "getSharedNpmArgs" function.

Reply:
Trouble is that there's a mix of args and environment variables. I wasn't sure how to combine the two in a logical manner.

Reply:
Never mind, figured it out--after changing the args to `string[]`, both can be passed to a shared method and updated.